### PR TITLE
#164139679 Set up swagger for API documentation

### PR DESCRIPTION
**Pull request by codinger41**
#### What does this PR do?
It sets up swagger for API documentation.

#### Description of Task to be completed?
It sets up swagger for API documentation and renders the UI on `${HOST_URL}/api-docs`

#### How should this be manually tested?
- Run `npm run start:dev` to start the application.
- Visit `${HOST_URL}/api-docs` to view the documentation UI.

#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
#164139679
#### Screenshots (if appropriate)
screenshot 2019-03-04 at 2 01 44 pm 1

#### Questions:
N/A

### CODE REVIEW

File: doc.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    "termsOfService": "http://swagger.io/terms/",
+    "contact": {
+      "email": "vidar.ah2019@gmail.com"
+    },
+    "license": {
+      "name": "GNU General Public License",
+      "url": "https://www.gnu.org/licenses/gpl-3.0.en.html"
+    }
+  },
+  "host": "authors-haven-staging.herokuapp.com",
```
Comment (youngestdj) [inline doc.json]: URL should be repo-name-staging.herokuapp.com for staging and repo-name.herokuapp.com for production.

Comment (youngestdj) [inline doc.json]:
```suggestion
  "host": "vidar-ah-backend-staging.herokuapp.com",
```

File: package.json [hunk 2]
```diff
@@ -5,16 +5,22 @@
   "main": "index.js",
   "scripts": {
     "coverage": "npm run db-migrate && mocha --reporter mochawesome --timeout=20000 --require babel-register --exit",
-    "db-migrate": "sequelize db:migrate:undo:all && sequelize db:migrate",
+    "db-migrate": "node_modules/.bin/sequelize db:migrate:undo:all && sequelize db:migrate",
```
Comment (9jaswag) [inline package.json]: this should not be referenced from node modules. The sequelize cli in dependencies should handle this.

**Review by 9jaswag — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -5,16 +5,22 @@
...
     "lint": "eslint routes models config test index.js",
     "test": "npm run db-migrate && mocha test/index.js --timeout=20000 --require babel-register --exit",
-    "start": "sequelize db:migrate && babel-node index.js",
+    "start": "node_modules/.bin/sequelize db:migrate && babel-node index.js",
```
Comment (9jaswag) [inline package.json]: babel-node should not be used in production. The code should be "transpiled" and run by Node

**Review by 9jaswag — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (9jaswag) [reply package.json]: Also, in production, we don't want to drop and run all migrations whenever we deploy to production. What happens to our data then? There's a way to only run migrations only when there's a migration to be run. Find out how. @leksyib

Comment (codinger41) [reply package.json]: Thanks, @9jaswag  will do.

File: .gitignore [hunk 4]
```diff
@@ -42,3 +42,4 @@ mochawesome-report
 
 # Ignore yarn generated files
 yarn.lock
+dist/
```
Comment (9jaswag) [inline .gitignore]: fix EOL issue

**Review by 9jaswag — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 4 above)
Comment (codinger41) [reply .gitignore]: Fixed

**Review by 9jaswag — APPROVED** [review]

**Pull request merged**